### PR TITLE
core-svc-feat(infographic-banners): add endpoint update sequence

### PR DESCRIPTION
**Pull request by ayocodingit**
## Overview

- core-svc-feat(infographic-banners): add endpoint update sequence

## Evidence
title: core-svc-feat(infographic-banners): add endpoint update sequence
project: Portal Jabar
participants: @ayocodingit @rachadiannovansyah @rhmnmbr @samudra-ajri @iqbal167

### CODE REVIEW

**Review by rhmnmbr — APPROVED** [review]

**Review by samudra-ajri — APPROVED** [review]

**Review by iqbal167 — APPROVED** [review]

File: core-service/src/modules/infographic-banner/usecase/infographic_banner_ucase.go [hunk 1]
```diff
@@ -112,3 +114,24 @@ func (i *infographicBannerUsecase) UpdateStatus(c context.Context, ID int64, bod
...
+
+func (i *infographicBannerUsecase) UpdateSequence(c context.Context, body *domain.UpdateSequenceInfographicBanner) (err error) {
+	ctx, cancel := context.WithTimeout(c, i.contextTimeout)
+	defer cancel()
+
+	tx, _ := i.infographicBannerRepo.GetTx(ctx)
+
+	for _, item := range body.Sequences {
+		if _, err = i.infographicBannerRepo.GetByID(ctx, item.ID, tx); err != nil {
+			err = errors.New(fmt.Sprintf("Your requested Item with ID %d is not found", item.ID))
```
Comment (rachadiannovansyah) [inline core-service/src/modules/infographic-banner/usecase/infographic_banner_ucase.go:126]: minor i think if this error has occurred on the use case layer, this error needs to logs to our elastic error log.

we can use GoLog/logrus helpers on this process

**Review by rachadiannovansyah — APPROVED** [review]

**Review by rachadiannovansyah — APPROVED** [review]

File: core-service/src/modules/infographic-banner/usecase/infographic_banner_ucase.go (hunk 1 above)
Comment (ayocodingit) [reply core-service/src/modules/infographic-banner/usecase/infographic_banner_ucase.go:126]: this is an error message override for the response, so we have message specific error with id, what do you suggest better keep in the case  Repository or Usecase?

Comment (ayocodingit) [reply core-service/src/modules/infographic-banner/usecase/infographic_banner_ucase.go:126]: for example like this 
image

Comment (rachadiannovansyah) [reply core-service/src/modules/infographic-banner/usecase/infographic_banner_ucase.go:126]: hoo this error is purely from repo ya, i think it is still relevant we catch it on the repository layer @ayocodingit

**Pull request merged**